crates/y-sweet-worker/src/lib.rs

```diff
@@ -123,15 +123,21 @@ async fn auth_doc(
         .map(|auth| auth.gen_doc_token(&doc_id, get_time_millis_since_epoch()));
 
     let url = if let Some(url_prefix) = &ctx.data.config.url_prefix {
-        let mut parsed = Url::parse(url_prefix).map_err(|_| Error::ConfigurationError)?;
+        let mut parsed = Url::parse(url_prefix).map_err(|_| Error::ConfigurationError {
+            field: "url_prefix".to_string(),
+            value: url_prefix.to_string(),
+        })?;
         match parsed.scheme() {
             "http" => parsed.set_scheme("ws").map_err(|_| Error::InternalError)?,
             "https" => parsed.set_scheme("wss").map_err(|_| Error::InternalError)?,
-            _ => return Err(Error::ConfigurationError),
+            _ => {
+                return Err(Error::ConfigurationError {
+                    field: "url_prefix".to_string(),
+                    value: url_prefix.to_string(),
+                })
+            }
         };
-        let result = parsed.join("/doc/ws").unwrap();
```

Comment:
what is the bug here exactly?

Reply:
`Url::join` works differently than a filesystem path join; if the string being joined starts with a `/`, it replaces the whole path. It aims to be consistent with how a browser would interpret a relative link from a website, in which case a leading `/` means to interpret it as relative to the _domain_ (technically, the authority); not the current path.

We could get the correct behavior here by:
- omitting `/` from the beginning of the joined string
- adding a `/` to the end of the URL (so that the project ID did not get replaced by the new path)

But I think it's clearer and less fragile just to use string manipulation.